flang/lib/Lower/ConvertVariable.cpp

```diff
@@ -1561,3 +1562,24 @@ void Fortran::lower::instantiateVariable(AbstractConverter &converter,
     instantiateLocal(converter, var, symMap);
   }
 }
+
+void Fortran::lower::mapCallInterfaceSymbols(
+    AbstractConverter &converter, const Fortran::lower::CallerInterface &caller,
+    SymMap &symMap) {
+  Fortran::lower::AggregateStoreMap storeMap;
+  for (auto var : Fortran::lower::pft::buildFuncResultDependencyList(
+           caller.getResultSymbol())) {
+    if (var.isAggregateStore()) {
+      instantiateVariable(converter, var, symMap, storeMap);
+    } else {
+      const Fortran::semantics::Symbol &sym = var.getSymbol();
+      // Do not instantiate the function result itself. It will be allocated
+      // later.
+      if (sym.IsFuncResult())
```

Comment:
Well, I was going to reply no, but then, internal procedure... I think it is possible in this case, given the result of the internal procedure may depend on host associated symbols including the result of the host functions.

```
function foo() result(res)
  implicit none
  integer :: res, i
  res = 10
  res = minval(sub_foo())
contains
  function sub_foo()
    integer :: sub_foo(res)
    integer, external :: ifunc
    sub_foo = [(ifunc(i), i = 1, res, 1)]
  end function
end function
```

I fixed it by preventing `buildFuncResultDependencyList` to return the result symbol itself since we do not care about it.
I also prevented the getPlacedMLIRArgFor to be called for dummy argument symbols of a host functions.